### ISSUE TITLE
Installation - Required PHP modules

### ISSUE THREAD

**Issue by ConfaLo** [Cacti/cacti#2665]
Hi, 
I have a problem with php check during installation process.
I run cacti on Debian 9 with apache 2, the installer shows me "cli - no" for all php modules.

![Annotazione 2019-05-02 104223](https://user-images.githubusercontent.com/26740894/57066846-a98a0100-6ccd-11e9-9756-0a0393edd153.png)

The command "php -m" reports them correctly:

```
[PHP Modules]
bcmath
calendar
Core
ctype
date
dom
exif
fileinfo
filter
ftp
gd
gettext
gmp
hash
iconv
json
ldap
libxml
mbstring
mysqli
mysqlnd
openssl
pcntl
pcre
PDO
pdo_mysql
Phar
posix
readline
Reflection
session
shmop
SimpleXML
snmp
sockets
SPL
standard
sysvmsg
sysvsem
sysvshm
tokenizer
wddx
xml
xmlreader
xmlwriter
xsl
Zend OPcache
zlib

[Zend Modules]
Zend OPcache
```

while executing "cli_check.php" on cacti installation dir I receive this report:
`{"ctype":{"cli":true,"web":false},"PDO":{"cli":true,"web":false},"zlib":{"cli":true,"web":false},"xml":{"cli":true,"web":false},"standard":{"cli":true,"web":false},"spl":{"cli":true,"web":false},"sockets":{"cli":true,"web":false},"simplexml":{"cli":true,"web":false},"session":{"cli":true,"web":false},"pdo_mysql":{"cli":true,"web":false},"pcre":{"cli":true,"web":false},"date":{"cli":true,"web":false},"openssl":{"cli":true,"web":false},"mbstring":{"cli":true,"web":false},"ldap":{"cli":true,"web":false},"json":{"cli":true,"web":false},"hash":{"cli":true,"web":false}, "gmp":{"cli":true,"web":false},"gd":{"cli":true,"web":false},"filter":{"cli":true,"web":false},"posix":{"cli":true,"web":false}}`

I can't find  where's the problem, I'm not a PHP expert, is there any missing configuration?

Thanks

**Comment by netniV** [comment]
That would suggests that the cli_check did find the results to be true, so should be reflecting that in the installation step.

What version is this?

**Comment by ConfaLo** [comment]
PHP is version 7.0.33 released with Debian, Cacti is a clone of the last version from github

**Comment by bmfmancini** [comment]
check to be sure that apache has  all of the modules loaded

Here is a snapshot of my dev install

sean@cacti-dev:~$ apache2ctl -M
AH00558: apache2: Could not reliably determine the server's fully qualified domain name, using 127.0.1.1. Set the 'ServerName' directive globally to suppress this message
Loaded Modules:
 core_module (static)
 so_module (static)
 watchdog_module (static)
 http_module (static)
 log_config_module (static)
 logio_module (static)
 version_module (static)
 unixd_module (static)
 access_compat_module (shared)
 alias_module (shared)
 auth_basic_module (shared)
 authn_core_module (shared)
 authn_file_module (shared)
 authz_core_module (shared)
 authz_host_module (shared)
 authz_user_module (shared)
 autoindex_module (shared)
 deflate_module (shared)
 dir_module (shared)
 env_module (shared)
 filter_module (shared)
 mime_module (shared)
 mpm_prefork_module (shared)
 negotiation_module (shared)
 php7_module (shared)
 setenvif_module (shared)
 status_module (shared)
sean@cacti-dev:~$ 

if this is a fresh install make sure to restart apache just to be sure it has picked up all of the packages

**Comment by ConfaLo** [comment]
apache configuration seems ok: 

```
apache2ctl -M

Loaded Modules:
 core_module (static)
 so_module (static)
 watchdog_module (static)
 http_module (static)
 log_config_module (static)
 logio_module (static)
 version_module (static)
 unixd_module (static)
 access_compat_module (shared)
 alias_module (shared)
 auth_basic_module (shared)
 authn_core_module (shared)
 authn_file_module (shared)
 authz_core_module (shared)
 authz_host_module (shared)
 authz_user_module (shared)
 autoindex_module (shared)
 deflate_module (shared)
 dir_module (shared)
 env_module (shared)
 filter_module (shared)
 mime_module (shared)
 mpm_prefork_module (shared)
 negotiation_module (shared)
 php7_module (shared)
 reqtimeout_module (shared)
 setenvif_module (shared)
 status_module (shared)
```

**Comment by netniV** [comment]
OK.  Can you do the following for me?

```sql
replace into settings values ('log_install','5')
```

Remove any file matching `log/install-*.log` then run through the install and get to the point of your problem.  Once you cannot go any further, take a copy of the `install-complete.log` to developers@cacti.net and I will take a look.

When you are done, you can remove the above setting by using
```sql
delete from settings where name = 'log_install'
```

**Comment by cigamit** [comment]
Any feedback?  Is you are still experiencing this issue, please update to the latest develop and have at it again.

**Comment by ConfaLo** [comment]
I tried to to clone again cacti from master repo, and finally all modules have been recognized!
I have probably cloned first time from develop branch, where there is something wrong with php checks. I Sent logs through email as requested by @netniV, hope can be useful  to investigate potential bugs.
Thanks for support!

**Comment by cjwbath** [comment]
I saw this issue when trying to install against a database imported from another server. In my case it was because the PHP binary was now in a different location to the _path_php_binary_ value in the database _settings_ table. Because of the way `utility_php_extensions()` shells out then silences errors from `json_decode`, if there was an error running PHP we just get an empty string. Once all the rest of the extension array processing is then done, it just looks to the GUI like none of the modules is installed for the CLI.

Once I did `update settings set value = '/usr/bin/php' where name = 'path_php_binary';` in the DB all was good. I suggest try to check the validity of that configuration setting earlier in the install process.
